content/articles/how-to-create-nlp-application-with-flair/index.md

```diff
@@ -0,0 +1,424 @@
+
+### Prerequisites
+1. You must have a good knowledge of [python](https://www.python.org/) and have it installed in your machine.
+2. You must have a good understanding of machine learning modeling.
+3. You must have a working knowledge of [Pandas](https://numpy.org/).
+4. You must have a working knowledge of [Numpy](https://numpy.org/).
+5. Use [Google Colab](https://research.google.com/) or [Jupyter Notebook](https://jupyter.org/). In this tutorial, we will be using Google Colab in building our model.
+
+### How to install Flair
+Since we are using [Google Colab](https://research.google.com/), use the following command to install flair:
+
+```python
+!pip install flair
+```
+After you have successfully installed Flair we can now start exploring Flair.
+
+### Loading the Flair package
+To load Flair so that we can start using it, we use the following command
+
+```python
+import flair
+```
+
+### Exploring Flair
+Exploring Flair enables us to see the methods and attributes available in this library, these methods will help us in building our text classification model.
+
+```python
+dir(flair)
+```
+
+Output of available methods and attributes:
+
+```bash
+['AnnealOnPlateau',
+ 'Path',
+ '__builtins__',
+ '__cached__',
+ '__doc__',
+ '__file__',
+ '__loader__',
+ '__name__',
+ '__package__',
+ '__path__',
+ '__spec__',
+ '__version__',
+ 'cache_root',
+ 'data',
+ 'datasets',
+ 'device',
+ 'embedding_storage_mode',
+ 'embeddings',
+ 'file_utils',
+ 'logger',
+ 'logging',
+ 'models',
+ 'nn',
+ 'optim',
+ 'os',
+ 'tokenization',
+ 'torch',
+ 'trainers',
+ 'training_utils',
+ 'visual']
+```
+
+The methods shown above such as `embeddings`, `data`, `models`, and `trainers` will be in building our model.
+`embeddings`- We shall use this to perform word embeddings.
+`models`- This will be used to specify the type of model we will be building.
+`trainers`- These methods will be used to train our model.
+
+### Prepare the dataset
+To start with data preparation, we need first to import the exploratory data analysis packages(EDA) to be used in data manipulation and analysis.
+
+#### Importing exploratory data analysis (EDA) packages
+The exploratory data analysis packages such as [Pandas](https://pandas.pydata.org/) and [Numpy](https://numpy.org/) will be used to reading and analysis of our data to get the characteristics available in our data.
+
+Let's import `pandas` and `numpy` into our Colab.
+
+```python
+import pandas as pd
+import numpy as np
+```
+
+### Dataset
+The dataset used consists of a collection of offensive and non-offensive words. The dataset will be used to train our model so that it can be used to predict if a text is offensive or not.
+
+[CSV File of our data](https://drive.google.com/file/d/1FpEnJkw7XY3cdP1NeUzRRKGqNYsQ3Frz/view?usp=sharing)
+Download the dataset from the above link and name the file as `offensive_and_non_offensive_dataset.csv`.
+
+#### Using Pandas to read dataset
+`Pandas` will be used to import our dataset.
```

Comment:
The Pandas library is used to import or read our data set? This sentence is not clear